cabal-install/Distribution/Client/Dependency.hs

```diff
@@ -699,13 +699,16 @@ validateSolverResult :: Platform
                      -> SolverInstallPlan
 validateSolverResult platform comp indepGoals pkgs =
     case planPackagesProblems platform comp pkgs of
-      [] -> case SolverInstallPlan.new indepGoals index of
+      [] -> case SolverInstallPlan.new indepGoals graph of
               Right plan     -> plan
               Left  problems -> error (formatPlanProblems problems)
       problems               -> error (formatPkgProblems problems)
 
   where
-    index = Graph.fromList pkgs
+    --FIXME: the solver produces large numbers of duplicate but identical
+    -- instances, so we currently eliminate duplicates here, and the
+    -- duplicate check below has to cope with there being identical instances.
+    graph = Graph.fromDistinctList (ordNubBy Graph.nodeKey pkgs)
```

Comment:
So given that the fixes to solver have landed, I guess that this can be removed now?